### PR TITLE
add axtchain tool from ucsc

### PR DESCRIPTION
**Pull request by martenson**
got test data down to:

```
-rw-r--r--@ 1 staff    17K Aug 18 12:22 hg38.chrM.fasta
-rw-r--r--@ 1 staff    32K Aug 18 10:25 hg38.chrM.mm39.chrM.axt
-rw-r--r--@ 1 staff   1.2K Aug 18 12:29 hg38.chrM.mm39.chrM.chain
-rw-r--r--@ 1 staff    16K Aug 18 12:22 mm39.chrM.fasta
```

FOR CONTRIBUTOR:
* [x] I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] License permits unrestricted use (educational + commercial)
* [x] This PR adds a new tool or tool collection
* [ ] This PR updates an existing tool or tool collection
* [ ] This PR does something else (explain below)

### CODE REVIEW

File: tools/ucsc_tools/axtchain/axtchain.xml [hunk 1]
```diff
@@ -0,0 +1,92 @@
+<tool id="axtchain" name="axtChain" version="@TOOL_VERSION@+galaxy0" profile="21.05" license="MIT">
```
Comment (bgruening) [inline tools/ucsc_tools/axtchain/axtchain.xml:1]:
```suggestion
<tool id="ucsc_axtchain" name="axtChain" version="@TOOL_VERSION@+galaxy0" profile="21.05" license="MIT">
```

File: tools/ucsc_tools/axtchain/axtchain.xml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                <option value="linear_gap_file">Provide a dataset with linear gap costs</option>
+            </param>
+            <when value="medium" />
+            <when value="loose" />
+            <when value="linear_gap_file">
+                <param name="lineargap_input" type="data" format="tabular" label="Read the linear gap costs from" help="See help for a sample file"/>
+            </when>
+        </conditional>
+
+        <param argument="-minScore" type="integer" optional="true" label="Minimum score for chain" help="Default 1000" />
```
Comment (bgruening) [inline tools/ucsc_tools/axtchain/axtchain.xml:53]: maybe add a min= value?

**Review by bgruening — COMMENTED** [review]
Looks great to me. Just two smaller comments.

**Comment by bgruening** [comment]
@martenson because @jmchilton is brewing something and you seem to be familiar now with the UCSC formats. Could you please add a `description` to the UCSC datatypes in https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/config/sample/datatypes_conf.xml.sample#L1008

File: tools/ucsc_tools/axtchain/axtchain.xml (hunk 2 above)
Comment (martenson) [reply tools/ucsc_tools/axtchain/axtchain.xml]: there is no min I am aware of, the software accepts negative score values

**Review by bgruening — APPROVED** [review]

**Pull request merged**